samples/graph-activity-feed/nodejs/static/scripts/auth-delegated.js

```diff
@@ -0,0 +1,108 @@
+            microsoftTeams.authentication.getAuthToken({
+                successCallback: (result) => {
+                    resolve(result);
+                },
+                failureCallback: function (error) {
+                    reject("Error getting token: " + error);
+                }
+            });
+
+        });
+    }
+
+    // 2. Exchange that token for a token with the required permissions
+    //    using the web service (see /auth/token handler in app.js)
+    function getServerSideToken(clientSideToken) {
+        return new Promise((resolve, reject) => {
+            microsoftTeams.initialize();
+            microsoftTeams.getContext((context) => {
+                fetch('/auth/token', {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        'Accept': 'application/json'
+                    },
+                    body: JSON.stringify({
+                        tid: context.tid,
+                        token: clientSideToken
+                    }),
+                })
+                    .then((response) => {
+                        if (response.ok) {
+                            return response.json();
+                        } else {
+                            reject(response.error);
+                        }
+                    })
+                    .then((responseJson) => {
+                        if (responseJson.error) {
+                            reject(responseJson.error);
+                        } else {
+                            const serverSideToken = responseJson;
+                            localStorage.setItem("accessToken", serverSideToken);
+                            resolve(serverSideToken);
+                        }
+                    });
+            });
+        });
+    }
+
+    // Show the consent pop-up
+    function requestConsent() {
+        return new Promise((resolve, reject) => {
+            microsoftTeams.authentication.authenticate({
+                url: window.location.origin + "/auth/auth-start",
+                width: 600,
+                height: 535,
+                successCallback: (result) => {
+                    let data = localStorage.getItem(result);
+                    localStorage.removeItem(result);
+                    resolve(data);
+                },
+                failureCallback: (reason) => {
+                    reject(JSON.stringify(reason));
+                }
+            });
+        });
+    }
+
+    // In-line code
+    getClientSideToken()
+        .then((clientSideToken) => {
+            return getServerSideToken(clientSideToken);
+        })
+        .catch((error) => {
+            if (error === "invalid_grant") {
+                console.log(`Error: ${error} - user or admin consent required`);
+                // Display in-line button so user can consent
+                let button = display("Consent", "button");
```

Comment:
where are we using this button? It might come as undefined, please test this scenario